src/main/java/htsjdk/variant/vcf/VCFCodec.java

```diff
@@ -72,91 +65,68 @@
-            }
-            else {
-                throw new TribbleException.InvalidHeader("We never saw the required CHROM header line (starting with one #) for the input VCF file");
-            }
+    public boolean canDecodeVersion(final VCFHeaderVersion targetHeaderVersion) {
+        return targetHeaderVersion == VCFHeaderVersion.VCF4_0 ||
+                targetHeaderVersion == VCFHeaderVersion.VCF4_1 ||
+                targetHeaderVersion == VCFHeaderVersion.VCF4_2;
+    }
 
-        }
-        throw new TribbleException.InvalidHeader("We never saw the required CHROM header line (starting with one #) for the input VCF file");
+    @Override
+    public boolean canDecode(final String potentialInput) {
+        // TODO: this will succeed on 4.3 files since it only looks as far as ..."##fileformat=VCFv4"
+        return canDecodeFile(potentialInput, VCF4_MAGIC_HEADER);
     }
 
     /**
-     * parse the filter string, first checking to see if we already have parsed it in a previous attempt
-     *
-     * @param filterString the string to parse
-     * @return a set of the filters applied or null if filters were not applied to the record (e.g. as per the missing value in a VCF)
+     * Handle reporting of duplicate filter IDs
+     * @param duplicateFilterMessage
      */
     @Override
-    protected List<String> parseFilters(final String filterString) {
-        // null for unfiltered
-        if ( filterString.equals(VCFConstants.UNFILTERED) )
-            return null;
-
-        if ( filterString.equals(VCFConstants.PASSES_FILTERS_v4) )
-            return Collections.emptyList();
-        if ( filterString.equals(VCFConstants.PASSES_FILTERS_v3) )
-            generateException(VCFConstants.PASSES_FILTERS_v3 + " is an invalid filter name in vcf4", lineNo);
-        if (filterString.isEmpty())
-            generateException("The VCF specification requires a valid filter status: filter was " + filterString, lineNo);
-
-        // do we have the filter string cached?
-        if ( filterHash.containsKey(filterString) )
-            return filterHash.get(filterString);
-
-        // empty set for passes filters
-        final List<String> fFields = new LinkedList<String>();
-        // otherwise we have to parse and cache the value
-        if ( !filterString.contains(VCFConstants.FILTER_CODE_SEPARATOR) )
-            fFields.add(filterString);
-        else
-            fFields.addAll(Arrays.asList(filterString.split(VCFConstants.FILTER_CODE_SEPARATOR)));
+    protected void reportDuplicateFilterIDs(final String duplicateFilterMessage) {
+        // older versions of htsjdk have been silently dropping these for a while, but we can at least warn
+        if (VCFUtils.getVerboseVCFLogging()) {
+            logger.warn(duplicateFilterMessage);
+        }
+    }
 
-        filterHash.put(filterString, Collections.unmodifiableList(fFields));
+    public void reportDuplicateInfoKeyValue(final String key, final String infoLine) {}
```

Comment:
Thanks for the explanation, @cmnbroad! Maybe a comment about this in the method will be good for discussion and knowing what is happening...